File: catalog/backends/python-django/patterns/ssrf_example_route.py
import requests as http_client
from django.http import JsonResponse

# Pattern: Code Security — SSRF via unvalidated URL parameter
# WARNING: intentionally vulnerable for IAST demo
# Adapt: use a domain-appropriate endpoint name


def fetch_url(request):
    url = request.GET.get("url", "")
    try:
        resp = http_client.get(url, timeout=5)
        return JsonResponse({"body": resp.text})
    except Exception as e:
        return JsonResponse({"error": str(e)}, status=500)

File: catalog/backends/node-express/patterns/ssrf_example_route.js
// Pattern: Code Security — SSRF via unvalidated URL parameter
// WARNING: intentionally vulnerable for IAST demo
// Adapt: use a domain-appropriate endpoint name

async function fetchUrl(req, res) {
  const url = req.query.url || '';
  try {
    const resp = await fetch(url);
    const body = await resp.text();
    res.send(body);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
}

module.exports = { fetchUrl };
